browser/js/products/products.js

```diff
@@ -0,0 +1,38 @@
+app.config(function ($stateProvider) {
```

Comment:
I would recommend separating this logic out. Having all of your states and controllers for multiple places in a single file can get long and frustrating. Given that it all get compiled into one giant file anyway, you might as well modularize the code so that everything is nice and readable.

That can either be putting all states in one file and all controllers in another, or maybe the `products` state and controller together and the `product` state and controller together.